miio/push_server.py

```diff
@@ -0,0 +1,401 @@
+import asyncio
+import calendar
+import datetime
+import logging
+import socket
+import struct
+from json import dumps
+from random import randint
+from typing import Any, Optional
+
+import attr
+
+from .device import Device
+from .protocol import Message, Utils
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVER_PORT = 54321
+FAKE_DEVICE_ID = "120009025"
+FAKE_DEVICE_MODEL = "chuangmi.plug.v3"
+HELO_BYTES = bytes.fromhex(
+    "21310020ffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+)
+
+
+@attr.s(auto_attribs=True)
+class EventInfo:
+    """Event info to register to the push server."""
+
+    action: str  # user friendly name of the event, can be set arbitrarily and will be received by the server as the name of the event
+    extra: str  # the identification of this event, this determines on what event the callback is triggered
+    event: Optional[str] = None  # defaults to the action
+    command_extra: str = ""
+    trigger_value: Optional[Any] = None
+    trigger_token: str = ""
+    source_sid: Optional[str] = None  # Normally not needed and obtained from device
+    source_model: Optional[str] = None  # Normally not needed and obtained from device
+    source_token: Optional[str] = None  # Normally not needed and obtained from device
+
+
+def calculated_token_enc(token):
+    token_bytes = bytes.fromhex(token)
+    encrypted_token = Utils.encrypt(token_bytes, token_bytes)
+    encrypted_token_hex = encrypted_token.hex()
+    return encrypted_token_hex[0:32]
+
+
+class PushServer:
+    """Async UDP push server using a fake miio device registered to a real miio
+    device."""
```

Comment:
Because the server is async you need quite a bit of code, the alarm_trigger example in the PR discription is basically the simplest example I could think off, I think it is quite long to put in the docstring.

I think it is best to write something in the docs, but I dont know where and in what format...
Could you make the required file and copy the alarm_trigger example in there?
I will then add some extra comments for clarification.

Reply:
I did add a partial example to the docstring, but it does not have all the proper async stuff included, just the steps that need to be taken.

Reply:
Okay, I added some preliminary documentation (and todos) just to give you an idea on how to do that. It is very rough, but hope it gets you started. I'm not sure if it makes sense to have two separate, but rather similar examples in the docs. Feel free to adjust as you see fit and let me know :-)

To test your changes locally, just go to the docs directory and execute `make html` to generate the docs you can then view in your browser. The docs are written in restructured text, https://docutils.sourceforge.io/docs/user/rst/quickref.html gives a brief overview of its features.

Reply:
I think the documentation is complete now.
Could you maybe have a look and make the syntax proper again?
I am really not into the restructured text format....